diff --git a/templates/nsis/include/extractAppPackage.nsh b/templates/nsis/include/extractAppPackage.nsh
--- a/templates/nsis/include/extractAppPackage.nsh
+++ b/templates/nsis/include/extractAppPackage.nsh
@@ -96,3 +96,13 @@
   SetOutPath "$PLUGINSDIR\7z-out"
-  Nsis7z::Extract "${FILE}"
+  !ifmacrodef customInstallerExtractStart
+    !insertmacro customInstallerExtractStart
+  !endif
+  !ifmacrodef customInstallerExtract
+    !insertmacro customInstallerExtract "${FILE}"
+  !else
+    Nsis7z::Extract "${FILE}"
+  !endif
+  !ifmacrodef customInstallerCopyStart
+    !insertmacro customInstallerCopyStart
+  !endif
   Pop $R0
@@ -107,3 +117,7 @@
     # Attempt to copy files in atomic way
-    CopyFiles /SILENT "$PLUGINSDIR\7z-out\*" $OUTDIR
+    !ifmacrodef customInstallerCopyFiles
+      !insertmacro customInstallerCopyFiles "$PLUGINSDIR\7z-out" "$OUTDIR"
+    !else
+      CopyFiles /SILENT "$PLUGINSDIR\7z-out\*" $OUTDIR
+    !endif
     IfErrors 0 DoneExtract7za
@@ -137,2 +151,5 @@
   DoneExtract7za:
+  !ifmacrodef customInstallerFilesReady
+    !insertmacro customInstallerFilesReady
+  !endif
 !macroend
